File: 양승민/입문/src/lec03/lec03Main.java
package lec03;

public class lec03Main {
    public static void main(String[] args) {

        // 1. 기본 타입
        // 암시적 변환
        int numberInt = 4;
        long numberLong = numberInt;
    }

    // 2. 타입 캐스팅
    public static void printAgeIfPerson(Object object) {
        if(object instanceof Person){
            Person person = (Person) object;
            System.out.println(person.getAge());
        }
    }

    // 4-1. String Interpolation
    Person person = new Person("양승민", 100);
    String log = String.format("사람의 이름은 %s이고 나이는 %s세 입니다", person.getName(), person.getAge());


    // 4-2. String Indexing
    String str = "ABCDE";
    char ch = str.charAt(1);

}

File: 양승민/입문/src/lec03/lec03Main.kt
package lec03

fun main() {

    // 1. 기본 타입
    val number1 = 3     // Int
    val number2 = 3L    // Long

    val number3 = 3.0f  // Float
    val number4 = 3.0   // Double

    // 명시적 변환
    val numberInt = 4
    val numberLong = numberInt.toLong()

    // 추가 예시
    val numberInt2 = 3
    val numberInt3 = 5
    val result = numberInt2 / numberInt3.toDouble()

    // 2. 타입 캐스팅
    fun printAgeIfPerson(obj: Any) {
        if(obj is Person) {
            val person = obj as Person
            println(person.age)
        }
    }

    // 개선
    fun printAgeIfPersonForSmart(obj: Any) {
        if(obj is Person) {
            println(obj.age)
        }
    }

    // 4-1. String Interpolation
    val person = Person("양승민", 100)
    val log = "사람의 이름은 ${person.name}이고 나이는 ${person.age}세 입니다"


    // 4-2. String Indexing
    val str = "ABCDE"
    val ch = str[1]

}